### DISCUSSION TITLE
What is the limit for the sync payload? How can we optimize?

### DISCUSSION THREAD

**Discussion by jericopulvera** [evoluhq/evolu#307] [Q&A] [answered]
For an application having a huge SQLite database for a single user is unlikely but possible. Is there a function we can call to display the SQLite database size? And a sync progress if the payload is huge? Also, maybe we can batch the sync? Manual syncing API also since syncing every single change seems like a waste of bandwidth when the database size is huge.

**Comment by steida** [comment, marked as answer]
>  Is there a function we can call to display the SQLite database size

No, there is no API in browsers for that yet. 

>  sync progress if the payload is huge?

No, but we can consider something API for that.

> Also, maybe we can batch the sync?

Why should we need that?

**Reply by jericopulvera** [reply]
Apologies, forget batch the sync. What I meant was to send multiple requests by parts to sync and bypass payload size limits, but I guess it does not solve the main issue. I just wanna address syncing problems that will arise due to the huge SQLite database size.

From what I can see right now is that the whole SQLite database data is sent and retrieved via HTTP to sync or restore the database, and that does not seem scalable and is very bandwidth expensive if most users have a huge SQLite database.

**Reply by steida** [reply]
We can improve that. Feel free to create an issue with the problem description. Thank you.
